Fix netavark compilation issues on FreeBSD #1

mnour opened the pull request:
- Add limited support for netlink on FreeBSD 13 and later, initially aiming to support functionality required by the netavark container networking manager

mnour committed:
Fix netavark compilation issues #1
- Initial changes
- Needs cleaning
- Needs adding support for FreeBSD 13 and
  15 as netlink support was first released in
  FreeBSD 13.2

dfr commented:
Looks good so far. Maybe re-word the commit message, e.g. 'Add limited support for netlink on FreeBSD 13 and later, initially aiming to support functionality required by the netavark container networking manager'

mnour committed:
Merge branch 'libc-0.2' into enhancement/1-fix-netavark-compilation-issues

mnour committed:
Fix netavark compilation issues #1
- Add Ethernet constants
- Add checks for netlink supported  FreeBSD versions
- Add code where it can support as many FreeBSD versions as needed

mnour committed:
Fix netavark compilation issues #1
- Minor formatting

mnour commented on src/unix/bsd/freebsdlike/freebsd/mod.rs:
```diff
@@ -4860,6 +4860,78 @@ pub const TFD_CLOEXEC: ::c_int = O_CLOEXEC;
+        /// sys/socket.h
+        pub const AF_NETLINK: ::c_int = 38;
+        pub const PF_NETLINK: ::c_int = AF_NETLINK;
+        
```
@dfr and @emaste open question to discuss during coming Monday call:
- According to [rtnetlink(4)](https://man.freebsd.org/cgi/man.cgi?query=rtnetlink&apropos=0&sektion=0&manpath=FreeBSD+14.1-RELEASE&arch=default&format=html) we only support:
1. AF_INET
2. AF_INET6

We need to add AF_BRIDGE in FreeBSD netlink/rtnetlink support.

mnour commented on src/unix/bsd/freebsdlike/freebsd/mod.rs:
```diff
@@ -4860,6 +4860,78 @@ pub const TFD_CLOEXEC: ::c_int = O_CLOEXEC;
+        pub const PF_NETLINK: ::c_int = AF_NETLINK;
+        
+        /// Not Supported (added for now just to compile)
+        pub const AF_BRIDGE: ::c_int = 7;
```
@dfr and @emaste open question to discuss during coming Monday call:
- According to [rtnetlink(4)](https://man.freebsd.org/cgi/man.cgi?query=rtnetlink&apropos=0&sektion=0&manpath=FreeBSD+14.1-RELEASE&arch=default&format=html) we only support:
1. AF_INET
2. AF_INET6

We may need to add AF_BRIDGE in FreeBSD netlink/rtnetlink support. So far I don't see any (direct) usage for/from netavark.

Cases:
- https://github.com/Project-Vacuum/nispor/blob/base/src/lib/query/inter_ifaces.rs#L90

mnour committed:
Fix netavark compilation issues on FreeBSD #1
- Add ARPHRD_ETHER

mnour committed:
Fix netavark compilation issues on FreeBSD #1
- Add RT_TABLE_MAIN const
- Update .gitignore
-# Don't track rustrover files

mnour committed:
Fix netavark compilation issues on FreeBSD #1
- Fix type
